ignition/service/queue.py

```diff
@@ -94,44 +96,47 @@ def __read_incoming_job(self, job_definition_str):
             return None
         return self.__handle_job(job_definition)
 
     def __handle_job(self, job_definition):
         requeue = False
         job_type = job_definition.get(self.JOB_TYPE_KEY, None)
+        job_id = job_definition.get(self.JOB_IDENTIFIER_KEY, None)
         if job_type is None:
-            logger.warning('Ignoring job received from queue without job_type: {0}'.format(job_definition))
+            logger.warning('Ignoring job received from queue without job_type for request id {0} and job id {1}'.format(job_definition.get('request_id'), job_id))
             return None
         else:
             job_handler = self.job_handlers.get(job_type, None)
             if job_handler is None:
-                logger.warning('No handler for job received from queue with job_type {0} (will re-queue)'.format(job_type))
+                logger.warning('No handler for job received from queue with job_type {0} and job id {1} (will re-queue)'.format(job_type, job_id))
                 requeue = True
             else:
-                logger.debug('Passing job to handler ({0}): {1}'.format(job_handler, job_definition))
+                logger.debug('Passing job to handler ({0}): for job type {1} and job id {2}'.format(job_handler, job_type, job_id))
                 try:
                     finished = job_handler(job_definition)
                 except Exception as e:
-                    logger.exception('Handling of job {0} returned an Exception, this task will not be re-queued. The error was: {1}'.format(job_definition, str(e)))
+                    logger.exception('Handling of job type {0} with job id {1} returned an Exception, this task will not be re-queued. The error was: {2}'.format(job_type, job_id, str(e)))
                     return None
                 if not finished:
-                    logger.debug('Handler marked job as incomplete, will re-queue: {0}'.format(job_definition))
+                    logger.debug('Handler marked job as incomplete, will re-queue job of job type: {0} with job id: {1}'.format(job_type, job_id))
                     requeue = True
         if requeue:
             self.queue_job(job_definition)
 
     def queue_job(self, job_definition):
-        logger.debug('Adding job to queue: {0}'.format(job_definition))
         if self.JOB_TYPE_KEY not in job_definition:
             raise ValueError('job_definition must have a job_type key')
         if job_definition[self.JOB_TYPE_KEY] is None:
             raise ValueError('job_definition must have a job_type value (not None)')
+        job_type = job_definition.get(self.JOB_TYPE_KEY, None)
+        if self.JOB_IDENTIFIER_KEY not in job_definition:
+            job_definition[self.JOB_IDENTIFIER_KEY] = str(uuid4())
+        logger.debug('Adding job to queue of job type: {0} with job id: {1}'.format(job_type, job_definition.get(self.JOB_IDENTIFIER_KEY, None)))
```

Comment:
The ":" may make this statement confusing as it's unsure where each section ends. So maybe this can just be: 
`Adding job to queue with type {0} and id {1}`

Reply:
updated